content/blog/2021-08-24-pre-defined-models-exp.md

```diff
@@ -0,0 +1,352 @@
+tags:
+  - MLOps
+  - Experiments
+  - Reproducibility
+  - DVC
+---
+
+## Intro
+
+There are plenty of machine learning models available that have been trained to
+solve one problem and the knowledge gained from that can be applied to a new,
+yet related problem. For example, a model like AlexNet has been trained on
+millions of images so you could potentially use this to classify cars, animals,
+or even people. This is called transfer learning and it can save a lot of time
+on developing a model from scratch.
+
+For us to take advantage of transfer learning, we can use fine-tuning to adopt
+the model to our new problem. In many cases, we start by replacing the last
+layer of the model. With the AlexNet example, this might mean the last layer was
+previously used to classify cars but our new problem is classifying animals.
+
+Even though we already have the bulk of the model defined, we'll still have to
+do some experimentation to determine whether we need to replace more layers in
+the model or if any other changes need to be made.
+
+In this post, we'll go through an example of fine-tuning
+[AlexNet](https://towardsdatascience.com/alexnet-the-architecture-that-challenged-cnns-e406d5297951)
+and
+[SqueezeNet](https://towardsdatascience.com/review-squeezenet-image-classification-e7414825581a)
+to classify bees and ants. We'll use DVC to handle experiments for us and we'll
+compare the results of both models at the end.
+
+## Initialize the pre-trained model
+
+We'll be fine-tuning the AlexNet model and the SqueezeNet model to classify
+images of bees and ants. You can find the project we're working with in
+[this repo](https://github.com/iterative/pretrained-model-demo).
+
+In the `pretrained_model_tuner.py` file, you'll find the code that defines both
+the AlexNet and SqueezeNet models. We start by initializing these models so we
+can get the number of model features and the input size we need for fine-tuning.
+
+Since the project has everything we need to initialize the models, we can start
+training and comparing the differences between them with the ants/bees dataset.
+Running experiments to get the best tuning for each model can make it difficult
+to see which changes led to a better result. That's why will be using DVC to
+track changes in the code and the data.
+
+## Add a DVC pipeline
+
+Before we begin training, let's set up a DVC pipeline. We'll do that by running
+the following command.
+
+```dvc
+$ dvc init
+```
+
+This creates the `.dvc/` directory for configuration, the default cache
+location, and other necessary files and directories for DVC to work.
+
+Make sure you set up a
+[virtual environment](https://python.readthedocs.io/en/stable/library/venv.html)
+and install the dependencies with:
+
+```bash
+$ pip install -r requirements.txt
+```
+
+Now that you have DVC initialized, this is a good place to commit your changes.
+
+```git
+$ git status
+$ git add .
+$ git commit -m "added DVC pipeline"
```

Comment:
this not a DVC pipeline yet ... it should be "DVC init"

Reply:
is it resolved? should we update the git commit message?

Reply:
I removed this section because we all agreed that we don't need to set up DVC in every blog post.